### PR TITLE
fix resize bug

### PR DESCRIPTION
**Pull request by BBuf**
I think the original intention here is to only take effect in the case of align_corners (because output_size = 1 and the divisor will be 0), but it affects non-align_corners too. For example: 

```python
input = torch.tensor(
        np.arange(1, 5, dtype=np.int32).reshape((1, 1, 2, 2)) )
m = torch.nn.Upsample(scale_factor=0.5, mode="bilinear")
of_out = m(input)
```

The result we expect should be [[[[2.5]]]] 

but pytorch get [[[[1.0]]]] which is different from OpenCV  and PIL, this pr try to fixed it。

### CODE REVIEW

**Comment by ezyang** [comment]
Could we get tests?

**Comment by ezyang** [comment]
@jbschlosser please reassign if you think you're not right reviewer

**Comment by jbschlosser** [comment]
Hey @BBuf, sorry for the delay! Is there an associated issue for this fix?

**Comment by BBuf** [comment]
> 
> 
> Hey @BBuf, sorry for the delay! Is there an associated issue for this fix?

There is no related issue, because users will hardly test the case where the sampling coefficient is less than 1. still hope this problem can be fixed.

File: aten/src/ATen/native/UpSample.h [hunk 1]
```diff
@@ -250,12 +250,16 @@ static inline scalar_t area_pixel_compute_scale(
...
+  if(align_corners){
+    if(output_size > 1){
+      return static_cast<scalar_t>(input_size - 1) / (output_size - 1);
+    }
+    else{
+      return static_cast<scalar_t>(0);
+    }
+  }
+  else{
+    return compute_scales_value<scalar_t>(scale, input_size, output_size);
```
Comment (jbschlosser) [inline aten/src/ATen/native/UpSample.h:262]: Please keep brace formatting consistent with elsewhere (spaces before curly braces and `} else {`)

File: aten/src/ATen/native/cuda/UpSample.cuh [hunk 2]
```diff
@@ -94,11 +94,16 @@ __host__ __forceinline__ static accscalar_t area_pixel_compute_scale(
...
+  if(align_corners){
+    if(output_size > 1){
+      return (accscalar_t)(input_size - 1) / (output_size - 1);
+    }
+    else{
+      return static_cast<accscalar_t>(0);
+    }
+  }
+  else{
+    return compute_scales_value<accscalar_t>(scale, input_size, output_size);
```
Comment (jbschlosser) [inline aten/src/ATen/native/cuda/UpSample.cuh:106]: Same here: please keep brace formatting consistent with elsewhere (spaces before curly braces and `} else {`)

**Review by jbschlosser — CHANGES_REQUESTED** [review]
Thanks for the fix! Please add tests for this case in `test/test_nn.py` next to the other upsampling tests.

File: aten/src/ATen/native/UpSample.h (hunk 1 above)
Comment (BBuf) [reply aten/src/ATen/native/UpSample.h:262]: okok

File: aten/src/ATen/native/cuda/UpSample.cuh (hunk 2 above)
Comment (BBuf) [reply aten/src/ATen/native/cuda/UpSample.cuh]: ok, I have added a test sample, can you see if there is any problem？

Comment (jbschlosser) [reply aten/src/ATen/native/cuda/UpSample.cuh]: Test looks okay to me

**Comment by jbschlosser** [comment]
cc @heitorschueroff do you want to take a look at this fix at all? It's related to interpolation

**Comment by heitorschueroff** [comment]
> cc @heitorschueroff do you want to take a look at this fix at all? It's related to interpolation

Sorry, I missed your cc. I'll take a look at it by tomorrow morning.

**Review by heitorschueroff — APPROVED** [review]
LGTM!

**Pull request closed without merging**